### PR TITLE
style: Add a fast-path for GeckoElement::is_root.

### PR DESCRIPTION
**Pull request by emilio**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19453)

### CODE REVIEW

**Comment by upsuper** [comment]
If `Gecko_IsRootElement` can always hit cache in `nsIDocument::GetRootElement()`, could we instead just remove that function and check `mCachedRootElement` of document directly?

**Comment by emilio** [comment]
It may not reach the cache when resolving styles from getComputedStyle on the main thread I believe.

**Comment by upsuper** [comment]
Hmm... okay. I guess.

File: components/style/gecko/wrapper.rs [hunk 1]
```diff
@@ -1917,6 +1917,15 @@ impl<'le> ::selectors::Element for GeckoElement<'le> {
 
     #[inline]
     fn is_root(&self) -> bool {
+        let parent_node = match self.as_node().parent_node() {
+            Some(parent_node) => parent_node,
```
Comment (upsuper) [inline components/style/gecko/wrapper.rs]: Actually, maybe just `Some(parent_node) if parent_node.is_document() => {},`?

**Pull request merged**